Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -191,10 +191,12 @@ ALWAYS_INLINE bool ContainerNode::removeNodeWithScriptAssertion(Node& childToRem
         ChildListMutationScope(*this).willRemoveChild(childToRemove);
     }
 
-    ASSERT_WITH_SECURITY_IMPLICATION(ScriptDisallowedScope::InMainThread::isEventDispatchAllowedInSubtree(childToRemove));
```

Comment:
This change is not great. Now we're not asserting that script execution is allowed whenever parser invokes this function. Also, we didn't update ContainerNode::removeAllChildrenWithScriptAssertion.

Reply:
OK — I can put together a PR with a patch to fix those problems

Reply:
> Also, we didn't update ContainerNode::removeAllChildrenWithScriptAssertion.

What change is needed to ContainerNode::removeAllChildrenWithScriptAssertion?

Reply:
It should be checking the flag just like ContainerNode::removeNodeWithScriptAssertion

Reply:
Addressing this and other issues in https://github.com/WebKit/WebKit/pull/25095.

Reply:
> Also, instead of suppressing assertion like this, we should do the closing of other details element in `didFinishInsertingNode` instead like this: …

Thanks — I’d never yet used `didFinishInsertingNode` in any of the patches I’ve worked on. And walking through it in the code now I see:

https://github.com/WebKit/WebKit/blob/a2066e9a79c56b8a83d98ef94bf6b818cea23eb1/Source/WebCore/dom/Node.h#L475-L478

That is, especially the _“insertedInto **MUST NOT** invoke scripts. Return NeedsPostInsertionCallback and implement didFinishInsertingNode instead to run scripts”_ comment.

So it’s clear to me now (and for if/when the next time comes that I need to do something like this in a later patch for another feature like this) how cases like this should be handled.